nixos/modules/services/x11/xserver.nix

```diff
@@ -251,11 +251,10 @@ in
 
       videoDrivers = mkOption {
         type = types.listOf types.str;
-        # !!! We'd like "nv" here, but it segfaults the X server.
-        default = [ "radeon" "cirrus" "vesa" "modesetting" ];
+        default = [ "amdgpu" "radeon" "nouveau" "modesetting" "fbdev" ];
```

Comment:
Just to summary the discussion of the other PR #100402:
- `amdgpu` and `radeon` works for both new and old AMD GPUs
- `nouveau` for NVIDIA GPUs
- `modesetting` for almost everything else
- `fbdev` is fallback if `modesetting` fails

`fbdev` seems a better fallback than `vesa` because VESA is only required for old BIOS systems (most GPUs should still support it though), while `fbdev` works almost every time there is a working framebuffer (i.e.: kernel TTY). It also seems compatible with EFI's framebuffer.

Reply:
Also important to note that using `vesa` requires an 8086 emulator running as root which prevents fixing https://github.com/NixOS/nixpkgs/issues/43984. https://lists.fedoraproject.org/archives/list/devel@lists.fedoraproject.org/thread/ESCVBW4KTXIAI4LZSDHC3IURXJW36DMW/

Reply:
I don't think setting `vesa` by default affects #43984 it unless you actually use `vesa` driver, but it is still a good point.

Reply:
(AFAIK, X.org only loads the driver if it actually needs it).